staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -132,26 +150,135 @@ func NewCloudNodeController(
 	return cnc, nil
 }
 
+// Run will sync informer caches and starting workers. 
 // This controller updates newly registered nodes with information
 // from the cloud provider. This call is blocking so should be called
 // via a goroutine
 func (cnc *CloudNodeController) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
+	defer cnc.workqueue.ShutDown()
 
 	// The following loops run communicate with the APIServer with a worst case complexity
 	// of O(num_nodes) per cycle. These functions are justified here because these events fire
 	// very infrequently. DO NOT MODIFY this to perform frequent operations.
 
+	// Wait for the caches to be synced before starting workers
+	klog.Info("Waiting for informer caches to sync")
+	if ok := cache.WaitForCacheSync(stopCh, cnc.nodesSynced); !ok {
+		klog.Errorf("failed to wait for caches to sync")
+		return
+	}
+
 	// Start a loop to periodically update the node addresses obtained from the cloud
-	wait.Until(func() { cnc.UpdateNodeStatus(context.TODO()) }, cnc.nodeStatusUpdateFrequency, stopCh)
+	go wait.Until(cnc.runWorker, cnc.nodeStatusUpdateFrequency, stopCh)
```

Comment:
I think we still want the periodic run of `UpdateNodeStatus` based on `cnc.nodeStatusUpdateFrequency` and `syncHandler` should actually only call `UpdateCloudNode` and not `UpdateNodeStatus`. So this should look like this instead:
```
go wait.Until(func() { cnc.UpdateNodeStatus(context.TODO()) }, cnc.nodeStatusUpdateFrequency, stopCh)
go wait.Until(c.runWorker, time.Second, stopCh)
```